File: Repo.Core/Models/Target.cs
using System;
using System.Collections.Generic;

namespace Repo.Core.Models;

public partial class Target
{
    public int ID { get; set; }

    public string Name { get; set; } = null!;

    public string Description { get; set; } = null!;

    public DateTime Start_Time { get; set; }

    public DateTime? Finish_Time { get; set; }

    public int? Tag_ID { get; set; }

    public virtual Tag? Tag { get; set; }
    
    public virtual ICollection<User> Users { get; set; } = new List<User>();
}

File: Repo.Frontend/src/app/modules/grade/interfaces/target.ts
export interface Target {
  id: number;
  name: string;
  description: string;
  start_Time: string;       // ISO DateTime from API, e.g. "2025-04-01T00:00:00"
  finish_Time?: string | null;
  tag?: string | null;
}

export type TargetMini = {
  name: string;
  description: string;
  start_Time: string;       // send as ISO (YYYY-MM-DD or full ISO)
  finish_Time?: string | null;
  tag?: string | null;
};
